### ISSUE TITLE
Unnecessary commit

### ISSUE THREAD

**Issue by Zordid** [MohammadRasool-Shaik/Kafka#4]
Here you do regular commits right after processing all records of one poll:
https://github.com/MohammadRasool-Shaik/Kafka/blob/9fc8b4b10fc07ca5b9e9f4cf14b7aee604e248a9/src/main/java/org/rash/kafka/consumer/ListnerConsumer.java#L59

There is little reason to do so, since *within* the upcoming next poll, right when the loop loops, the installed rebalance listener would do exactly the same thing, if you hadn't cleared the offsets here.

In other words: in this implementation of RebalanceListener and your consumer loop, the committing action in "onRevokePartitions" would actually never ever commit anything, the currentOffsets would *always* be empty as this has always just been done anyway here in this line.

Remember: the RebalanceListener's methods are always just invoked *within* your own poll call! Not at any other point in time.
